### ISSUE TITLE
CodeArtifact GetAuthorizationToken incorrectly parsing returned expiry timestamp

### ISSUE THREAD

**Issue by cloudshiftchris** [awslabs/aws-sdk-kotlin#737]
### Describe the bug

When invoking CodeArtifact.GetAuthorizationToken the returned expiry timestamp is incorrectly parsed.

### Expected behavior

Returned timestamp is correctly parsed.

### Current behavior

As shown in debug output below, the returned expiry timestamp of `"expiration":1.667042583E9` is incorrectly parsed by the SDK into a timestamp object with a toString() of `1970-01-01T00:00:01.667042Z`.  It appears the exponent is not being processed, resulting in processing the timestamp as 1.667 seconds (instead of 1.667E9).

```
val authorizationTokenResult = opts.codeArtifactClient.getAuthorizationToken {
    domain = codeArtifactUrl.domain
    durationSeconds = opts.codeArtifactTokenDuration.inWholeSeconds  // duration set to 12h (this worked file with AWS Java SDK v2)
}

2022-10-28T16:23:03.367-0700 [DEBUG] [httpTraceMiddleware] sdkRequestId: 7a765bd3-d06d-4738-9fae-593390545e15; service: codeartifact; operation: GetAuthorizationToken; - HttpRequest:
POST /v1/authorization-token?domain=****&duration=43200
Host: codeartifact.us-west-2.amazonaws.com
amz-sdk-invocation-id: 7a765bd3-d06d-4738-9fae-593390545e15
amz-sdk-request: attempt=1; max=3
User-Agent: aws-sdk-kotlin/0.16.4-beta os/macos/13.0 lang/kotlin/1.6.21
x-amz-user-agent: aws-sdk-kotlin/0.16.4-beta
X-Amz-Date: 20221028T232303Z
X-Amz-Security-Token: ***
Authorization: AWS4-HMAC-SHA256 Credential=***/20221028/us-west-2/codeartifact/aws4_request, SignedHeaders=amz-sdk-invocation-id;amz-sdk-request;host;x-amz-date;x-amz-security-token;x-amz-user-agent, Signature=97c4abb3e1c6ae47d66bfc2077ac394e6f60e58235806aa0e181be968521b9b9

... okhttp HTTP2/TLS/framing omitted ...

2022-10-28T16:23:03.740-0700 [DEBUG] [httpTraceMiddleware] sdkRequestId: 7a765bd3-d06d-4738-9fae-593390545e15; service: codeartifact; operation: GetAuthorizationToken; - HttpResponse:
HTTP 200: OK
content-length: 1558
content-type: application/json
date: Fri, 28 Oct 2022 23:23:03 GMT
x-amzn-requestid: 2b197944-5964-47d4-90ad-96fb5745ec4a

{"authorizationToken":"***","expiration":1.667042583E9}

logger.debug {
    "Obtained CodeArtifact token ${
    authorizationTokenResult.authorizationToken?.dropLast(
        15
    )
    }*****; expiry: ${authorizationTokenResult.expiration}"
}

2022-10-28T16:23:03.741-0700 [DEBUG] [cloudshift.gradle.settings.plugin.repository.codeartifact.tokenresolver.aws.AwsCodeArtifactTokenRepository] Obtained CodeArtifact token ****; expiry: 1970-01-01T00:00:01.667042Z
```

### Steps to Reproduce

Make a call to CodeArtifact.GetAuthorizationToken and view the result:

```
val authorizationTokenResult = opts.codeArtifactClient.getAuthorizationToken {
    domain = codeArtifactUrl.domain
    durationSeconds = opts.codeArtifactTokenDuration.inWholeSeconds  // duration set to 12h (this worked file with AWS Java SDK v2)
}
```

### Possible Solution

Unclear where in the SDK internals this parsing lives; also possible this is fixed in newer SDK versions (on 0.16.4-beta for Kotlin 1.6 compatibility).

### Context

Unable to directly use CodeArtifact token expiry.  As a workaround will set it to 1h (it's typically 12h).

### AWS Kotlin SDK version used

0.16.4-beta

### Platform (JVM/JS/Native)

JVM

### Operating System and version

MacOS 13.0

**Comment by cloudshiftchris** [comment]
Possibly the same as #665

**Comment by aajtodd** [comment]
Recreated on `0.16.4-beta`, confirmed this was indeed fixed and works in latest `0.17.9-beta`.

Closing as duplicate of #665
